addon/-private/system/store/finders.js

```diff
@@ -196,6 +196,11 @@ export function _queryRecord(adapter, store, typeClass, query) {
     var record;
     store._adapterRun(function() {
       var payload = normalizeResponseHelper(serializer, store, typeClass, adapterPayload, null, 'queryRecord');
+
+      assert("Expected the primary data returned by the serializer for a `queryRecord` response to be a single object but instead it was an array.", !Array.isArray(payload.data), {
+        id: 'ds.store.queryRecord-array-response'
+      });
+
```

Comment:
At first sight it seemed that was not necessary here, but I guess it handles the case of customs adapters/serializers, right ?